src/apm-cli.js

```diff
@@ -0,0 +1,244 @@
+const {spawn} = require('child_process');
+const path = require('path');
+const _ = require('underscore-plus');
+const colors = require('colors');
+const npm = require('npm');
+const yargs = require('yargs');
+const wordwrap = require('wordwrap');
+// Enable "require" scripts in asar archives
+require('asar-require');
+const config = require('./apm');
+const fs = require('./fs');
+const git = require('./git');
+
+function setupTempDirectory() {
+  const temp = require('temp');
+  // Resolve ~ in tmp dir atom/atom#2271
+  let tempDirectory = require('os').tmpdir();
+  temp.dir = path.resolve(fs.absolute(tempDirectory));
+  try {
+    fs.makeTreeSync(temp.dir);
+  } catch (e) {}
+  return temp.track();
+};
+
+setupTempDirectory();
+
+const commandClasses = [
+  require('./ci'),
+  require('./clean'),
+  require('./config'),
+  require('./dedupe'),
+  require('./develop'),
+  require('./disable'),
+  require('./docs'),
+  require('./enable'),
+  require('./featured'),
+  require('./init'),
+  require('./install'),
+  require('./links'),
+  require('./link'),
+  require('./list'),
+  require('./login'),
+  require('./publish'),
+  require('./rebuild'),
+  require('./rebuild-module-cache'),
+  require('./search'),
+  require('./star'),
+  require('./stars'),
+  require('./test'),
+  require('./uninstall'),
+  require('./unlink'),
+  require('./unpublish'),
+  require('./unstar'),
+  require('./upgrade'),
+  require('./view')
+];
+
+const commands = {};
+
+for (const commandClass of commandClasses) {
+  for (const name of commandClass.commandNames || []) {
+    commands[name] = commandClass;
+  }
+}
+
+function parseOptions(args) {
+  if (!args) args = [];
+  const options = yargs(args).wrap(Math.min(100, yargs.terminalWidth()));
+  options.usage("\nPulsar Package Manager powered by https://pulsar-edit.com\n\n  Usage: apm <command>\n\n  where <command> is one of:\n  " + (wordwrap(4, 80)(Object.keys(commands).sort().join(', '))) + ".\n\n  Run `apm help <command>` to see the more details about a specific command.");
```

Comment:
We can use the quote syntax here:

```js
  options.usage(`
```suggestion
  options.usage("\nPulsar Package Manager powered by https://pulsar-edit.com\n\n  Usage: apm <command>\n\n  where <command> is one of:\n  " + (wordwrap(4, 80)(Object.keys(commands).sort().join(', '))) + ".\n\n  Run `apm help <command>` to see the more details about a specific command.");
```
  Pulsar Package Manager powered by https://pulsar-edit.com
    Usage: apm <command>
    where <command> is one of:
    ${wordwrap(4, 80)(Object.keys(commands).sort().join(', '))}.
    Run \`apm help <command>\` to see the more details about a specific command.
  `)
```

Reply:
This will include all the spaces before the text starts. Therefor it would look like this and I think that is ugly.
```js
  options.usage(`
Pulsar Package Manager powered by https://pulsar-edit.com

  Usage: apm <command>

  where <command> is one of:
  ${wordwrap(4, 80)(Object.keys(commands).sort().join(', '))}.

  Run \`apm help <command>\` to see the more details about a specific command.\
  `);
```

Reply:
The multiline boring problem :roll_eyes: 
And using this style? 
```js
options.usage(
  "Pulsar Package Manager powered by https://pulsar-edit.com\n"+
  "\n"+
  "Usage: apm <command>\n"+
  "....."
```

Reply:
What do you think of this method?
```js
  options.usage([
    "",
    "Pulsar Package Manager powered by https://pulsar-edit.com",
    "",
    "  Usage: apm <command>",
    "",
    "  where <command> is one of:",
    `  ${wordwrap(4, 80)(Object.keys(commands).sort().join(', '))}.`,
    "",
    "  Run `apm help <command>` to see the more details about a specific command.",
  ].join("\n"));
```

Reply:
> The multiline boring problem 🙄 And using this style?
> 
> ```js
> options.usage(
>   "Pulsar Package Manager powered by https://pulsar-edit.com\n"+
>   "\n"+
>   "Usage: apm <command>\n"+
>   "....."
> ```

Oh, I didn't thought of that. Thanks